packages/pg-pool/test/background-socket-terminiation.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+const net = require('net')
+const co = require('co')
+const expect = require('expect.js')
+
+const describe = require('mocha').describe
+const it = require('mocha').it
+
+const Pool = require('../')
+
+describe('a socket disconnecting in the background', () => {
+  it('should leave the pool in a usable state', async () => {
+    const pool = new Pool({ max: 1 })
+    // just run any arbitrary query
+    const client = await pool.connect()
+    await client.query('SELECT NOW()')
+    // return the client
+    client.release()
+
+    // now kill the socket in the background
+    client.connection.stream.end()
+
+    // now try to query again, it should work
+    await pool.query('SELECT NOW()')
+    await pool.query('SELECT NOW()')
+    await pool.query('SELECT NOW()')
+    await pool.query('SELECT NOW()')
+
```

Comment:
Perhaps you should also add something like:

```
const client2 = await pool.connect();
try {
  // Verify that `client` was removed from the pool and replaced w/ a new client
  expect(client2).to.not.equal(client);
} finally {
  await client2.release();
}
```

Reply:
(Otherwise, it's possible that `client` just magically fixed itself through some other means.  So, adding this check helps to rule out that possibility)

Reply:
yah that's not a bad idea!  I'd like to hold off on merging this until I get some confirmation its actually fixing things in aws lambda.  It also unfortunately doesn't work w/ the native bindings yet so I'll need to somehow figure that out.

Reply:
Sounds good.

As I mentioned in [this thread](https://github.com/brianc/node-postgres/issues/2112#issuecomment-591040368), I suspect that both `"pg"` and `"knex"` were encountering the same issue within their respective pool implementations.  So, we'll most likely need to implement a similar fix on the `"knex"` side.